androidsdk/src/main/java/com/wootric/androidsdk/Wootric.java

```diff
@@ -452,6 +452,14 @@ public void setSurveyTypeScale(int surveyTypeScale) {
         settings.setCustomSurveyTypeScale(surveyTypeScale);
     }
 
+    /**
+     * Hides a survey if it’s displayed without generating a decline.
+     * If the delay time is still running stop will abort that survey from happening.
+     */
```

Comment:
Hey, @diegoserranoa before going through testing, can you please help me understand a few points. I see the end goals here for the `stop()` method as mentioned.
1. If the user is eligible, the point at which the survey is triggered to the point it's going to show survey. The time duration between those two points can be because of specially `wootric.setFirstSurveyDelay`. Therefore using `wootric.stop()` method after `wootric.survey()` - in that case should stop survey from showing at all right?
2. If the user is eligible and if the survey is already triggered also shown`. Therefore using `wootric.stop()` method after `wootric.survey()` - in that case it should dismiss the survey right? but as you mentioned it should not dismiss creating decline? Won't it be against the user experience flow or it's the requirement?

Thanks for working out on this, I will gradually test this out. Have eyes on this 🙏

Reply:
1. Yes, the calling stop should prevent the survey from showing up.
2. Yes, if the survey is shown, it should dismiss the survey. The JS SDK works in the same way so we're trying to be consistent here. It shouldn't create a decline since the user didn't really decline the survey.

## Tests

There isn't an specific way to test this. One thing I've been doing is using the [callbacks](https://docs.wootric.com/android/#survey-callbacks) so that way I'm aware of when the survey will show up. You can also try using a long time delay (e.g. 10 seconds). That way you can test calling stop before and after the survey has showed up.

Reply:
Cool work @diegoserranoa - works as expected. Just one question for clarification. Is it right that survey should stop showing when using `stop()` method even when setting `survey_immediately` to `true`.

Reply:
@ambarwootric correct. Survey immediately is for calling eligibility, the stop method is when the user is already eligible.